assets/src/dashboard/app/views/shared/templateGridView.js

```diff
@@ -62,32 +72,91 @@ function TemplateGridView({ pageSize, templates, templateActions }) {
     [templateActions]
   );
 
+  useGridViewKeys({
+    containerRef,
+    gridRef,
+    itemRefs,
+    isRTL,
+    currentItemId: activeGridItemId,
+    items: templates,
+  });
+
+  // when keyboard focus changes through FocusableGridItem immediately focus the CardPreviewContainer
+  useEffect(() => {
+    itemRefs.current?.[activeGridItemId]?.lastChild.focus();
+  }, [activeGridItemId]);
+
   return (
-    <GridContainer pageSize={pageSize}>
-      {templates.map((template) => (
-        <CardGridItem
-          key={template.id}
-          data-testid={`template-grid-item-${template.id}`}
-        >
-          <CardPreviewContainer
-            pageSize={pageSize}
-            story={template}
-            centerAction={{
-              targetAction: template.centerTargetAction,
-              label:
-                TEMPLATES_GALLERY_ITEM_CENTER_ACTION_LABELS[template.status],
-            }}
-            bottomAction={{
-              targetAction: targetAction(template),
-              label: __('Use template', 'web-stories'),
-            }}
-            containerAction={(e) =>
-              templateActions.handlePreviewTemplate(e, template)
-            }
-          />
-        </CardGridItem>
-      ))}
-    </GridContainer>
+    <div ref={containerRef}>
+      <GridContainer
+        pageSize={pageSize}
+        gridRef={gridRef}
+        ariaLabel={__('Available templates', 'web-stories')}
+      >
+        {templates.map((template) => {
+          const isActive = activeGridItemId === template.id;
+          const tabIndex = isActive ? 0 : -1;
+          return (
+            <CardGridItem
+              key={template.id}
+              data-testid={`template-grid-item-${template.id}`}
+              ref={(el) => {
+                itemRefs.current[template.id] = el;
+              }}
+            >
+              <FocusableGridItem
+                onFocus={() => {
+                  setActiveGridItemId(template.id);
+                }}
+                isSelected={isActive}
+                tabIndex={tabIndex}
+                aria-label={
+                  isActive
+                    ? sprintf(
+                        /* translators: %s: template title.*/
+                        __('%s (currently selected)', 'web-stories'),
+                        template.title
+                      )
+                    : template.title
+                }
+              />
+              <CardPreviewContainer
+                ariaLabel={
+                  isActive
+                    ? sprintf(
+                        /* translators: %s: template title.*/
+                        __('%s (currently selected)', 'web-stories'),
```

Comment:
Because we have two sibling elements with the same aria label it can read the same thing twice. Either we should wrap `CardPreviewContainer` with `FocusableGridItem` as a `div` and set the role to `button` or pass the props from the first to the second.

Reply:
great point, i was focusing on the keyboard functionality and hadn't noticed the duplication in the screen reader.  

I adjusted this so that the hidden focusable grid item that controls focus within grid items while using the keyboard now has a button title that says: "Press Enter to explore {template title} template". The _only_ time this button is read (at least when I enable my voice over and test it) is when you are in a template already and hit `Tab + Shift` til you get to the top of a card. The focus is automatically pushed to the preview overlay when the grid item id is updated, so it's only when there's no id to update that this button is a fall back. 

Then I adjusted the label on the entire preview container that serves as the true landing point of the grid item to read "Currently on {template title} template. And I added specific aria labels for each button inside the preview container to more accurately describe the button's action - namely so the user can know that they are going to get a new view with any of those interactions. 

They are: 
"Preview" => "Open preview"
"See Details" => "Go to template details" 
"Use template" => "Use this template"

This makes the screen reader experience a lot better, I think - let me know?